### PR TITLE
BUGZ-829 DEV-168 - domain backup/content management improvements

### PR DESCRIPTION
**Pull request by roxanneskelly**
BUGZ-829 - improve error reporting during restore of content

DEV-168 - show last installed archive information

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20531/HighFidelity-Beta-PR15978-783c615.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20531/HighFidelity-Quest-Beta-PR15978-783c615.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20531/HighFidelity-Beta-PR15978-783c615.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20531/HighFidelity-Beta-Interface-PR15978-783c615.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20531/HighFidelity-Beta-PR15978-783c615.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20531/HighFidelity-Beta-Interface-PR15978-783c615.dmg

File: domain-server/src/DomainServer.h [hunk 1]
```diff
@@ -194,6 +194,7 @@ private slots:
     QUrl oauthRedirectURL();
     QUrl oauthAuthorizationURL(const QUuid& stateUUID = QUuid::createUuid());
 
+    QString getWebSessionUsername(HTTPConnection* connection);
```
Comment (howard-stearns) [inline domain-server/src/DomainServer.h]: where is this defined?

Comment (roxanneskelly) [reply domain-server/src/DomainServer.h]: oopse, that's part of a stretch goal item.  Will remove

File: domain-server/resources/describe-settings.json [hunk 2]
```diff
@@ -1788,6 +1788,29 @@
...
+      "label": "Installed Archive",
+      "hidden": true,
+      "settings": [
+        {
+          "name": "filename",
+          "content_setting": true,
+          "default": ""
+        },
+        {
+          "name": "install_time",
```
Comment (howard-stearns) [inline domain-server/resources/describe-settings.json]: I think there was some back and forth in the spec, but it sounds like you were saying that we would (attempt to) have the install time, and when available, the creation time of the content set? I don't see the latter here.

File: domain-server/resources/describe-settings.json [hunk 3]
```diff
@@ -1788,6 +1788,29 @@
           "default": false
         }
       ]
+    },
+    {
+      "name": "installed_archive",
+      "label": "Installed Archive",
```
Comment (howard-stearns) [inline domain-server/resources/describe-settings.json]: I'd like to see the language generalized just a bit, to cover the other two cases: restore from backup, and inventory placement.

I _think_ that I do like "installed". (I think "restored" had also been discussed, but I like that "installed" is a bit more general.)

Maybe "Installed Content"?

File: domain-server/resources/describe-settings.json (hunk 2 above)
Comment (roxanneskelly) [reply domain-server/resources/describe-settings.json]: It's retained as part of the filename.

File: domain-server/resources/describe-settings.json [hunk 4]
```diff
@@ -1788,6 +1788,29 @@
...
+      "hidden": true,
+      "settings": [
+        {
+          "name": "filename",
```
Comment (howard-stearns) [inline domain-server/resources/describe-settings.json]: I'm not sure that I'm understanding, but I _think_ that the string "filename" is internal, and that the choice of one word or another is just whatever makes it clearer to developers.  As a developer, I'm a bit confused... I think that the value is the name that we hope will come from inside the .zip ideally - and is thus not a "filename". (If I understand correctly, it _might_ indeed be a filename, e.g., if we don't have a better info in, say, an "Upload Content" button usage of a .gz or .zip that predates this PR.)

File: domain-server/resources/describe-settings.json (hunk 3 above)
Comment (roxanneskelly) [reply domain-server/resources/describe-settings.json]: Still waiting for feedback from design on this

File: domain-server/resources/describe-settings.json (hunk 4 above)
Comment (roxanneskelly) [reply domain-server/resources/describe-settings.json]: This is an internal-only value that corresponds to the filename of the archive at creation time, which is composed of a type, a name, and a date.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20544/HighFidelity-Beta-PR15978-fe4334d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20544/HighFidelity-Beta-Interface-PR15978-fe4334d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20544/HighFidelity-Beta-PR15978-fe4334d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20544/HighFidelity-Beta-Interface-PR15978-fe4334d.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20544/HighFidelity-Beta-PR15978-fe4334d.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20544/HighFidelity-Quest-Beta-PR15978-fe4334d.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20551/HighFidelity-Beta-PR15978-5afae43.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20551/HighFidelity-Beta-Interface-PR15978-5afae43.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20551/HighFidelity-Beta-PR15978-5afae43.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20551/HighFidelity-Quest-Beta-PR15978-5afae43.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20551/HighFidelity-Beta-PR15978-5afae43.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20551/HighFidelity-Beta-Interface-PR15978-5afae43.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20569/HighFidelity-Beta-PR15978-22e7111.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20569/HighFidelity-Beta-Interface-PR15978-22e7111.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20569/HighFidelity-Beta-PR15978-22e7111.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20569/HighFidelity-Quest-Beta-PR15978-22e7111.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20569/HighFidelity-Beta-PR15978-22e7111.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20569/HighFidelity-Beta-Interface-PR15978-22e7111.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20581/HighFidelity-Beta-PR15978-130e68c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20581/HighFidelity-Quest-Beta-PR15978-130e68c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20581/HighFidelity-Beta-PR15978-130e68c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20581/HighFidelity-Beta-Interface-PR15978-130e68c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20581/HighFidelity-Beta-PR15978-130e68c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20581/HighFidelity-Beta-Interface-PR15978-130e68c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20594/HighFidelity-Beta-PR15978-b770ff7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20594/HighFidelity-Quest-Beta-PR15978-b770ff7.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20594/HighFidelity-Beta-PR15978-b770ff7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20594/HighFidelity-Beta-Interface-PR15978-b770ff7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20594/HighFidelity-Beta-PR15978-b770ff7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20594/HighFidelity-Beta-Interface-PR15978-b770ff7.exe

File: domain-server/src/DomainServer.cpp [hunk 5]
```diff
@@ -2601,14 +2601,16 @@ bool DomainServer::processPendingContent(HTTPConnection* connection, QString ite
         }
         QByteArray& _pendingUploadedContent = _pendingUploadedContents[sessionId];
         _pendingUploadedContent += dataChunk;
-        connection->respond(HTTPConnection::StatusCode200);
 
         if (itemName == "restore-file" || itemName == "restore-file-chunk-final" || itemName == "restore-file-chunk-only") {
             // invoke our method to hand the new octree file off to the octree server
-            QMetaObject::invokeMethod(this, "handleOctreeFileReplacement",
-                Qt::QueuedConnection, Q_ARG(QByteArray, _pendingUploadedContent), Q_ARG(QString, filename), Q_ARG(QString, QString()));
+            if(!handleOctreeFileReplacement(_pendingUploadedContent, filename, QString())) {
```
Comment (howard-stearns) [inline domain-server/src/DomainServer.cpp]: doh! coding standards!

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20601/HighFidelity-Beta-PR15978-908cb81.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20601/HighFidelity-Beta-Interface-PR15978-908cb81.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20601/HighFidelity-Beta-PR15978-908cb81.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20601/HighFidelity-Beta-Interface-PR15978-908cb81.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20601/HighFidelity-Beta-PR15978-908cb81.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20601/HighFidelity-Quest-Beta-PR15978-908cb81.apk)

**Comment by antoninahifi** [comment]
currently testing

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20609/HighFidelity-Beta-PR15978-03b146c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20609/HighFidelity-Beta-Interface-PR15978-03b146c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20609/HighFidelity-Beta-PR15978-03b146c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20609/HighFidelity-Quest-Beta-PR15978-03b146c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20609/HighFidelity-Beta-PR15978-03b146c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20609/HighFidelity-Beta-Interface-PR15978-03b146c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20613/HighFidelity-Beta-PR15978-701c457.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20613/HighFidelity-Beta-Interface-PR15978-701c457.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20613/HighFidelity-Beta-PR15978-701c457.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20613/HighFidelity-Beta-Interface-PR15978-701c457.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20613/HighFidelity-Beta-PR15978-701c457.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20613/HighFidelity-Quest-Beta-PR15978-701c457.apk)

**Pull request merged**